tests/imodels-clients-tests/src/unit/management/DefaultHttpRequestRetryPolicy.test.ts

```diff
@@ -0,0 +1,63 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { DefaultHttpRequestRetryPolicy, ExponentialBackoffAlgorithm, IModelsErrorImpl } from "@itwin/imodels-client-management/lib/base/internal";
+import { expect } from "chai";
+import sinon from "sinon";
+
+import { GetSleepDurationInMsParams, IModelsErrorCode } from "@itwin/imodels-client-management";
+
+import { createStub } from "../Stubs";
+
+describe("[Management] DefaultRetryPolicyProvider", () => {
+  let backoffAlgorithmStub: sinon.SinonStubbedInstance<ExponentialBackoffAlgorithm>;
+
+  before(() => {
+    backoffAlgorithmStub = createStub(ExponentialBackoffAlgorithm);
+  });
+
+  it("should use provided backoff algorithm when calculating sleep duration", () => {
+    // Arrange
+    const testedClass = new DefaultHttpRequestRetryPolicy(backoffAlgorithmStub);
+    const params: GetSleepDurationInMsParams = { retriesInvoked: 2 };
+    const expectedSleepDuration = 1000;
+    backoffAlgorithmStub.getSleepDurationInMs.withArgs(params.retriesInvoked).returns(expectedSleepDuration);
+
+    // Act
+    const actualSleepDuration = testedClass.getSleepDurationInMs(params);
+
+    // Assert
+    expect(actualSleepDuration).to.be.equal(expectedSleepDuration);
+    expect(backoffAlgorithmStub.getSleepDurationInMs.calledOnceWith(params.retriesInvoked)).to.be.true;
+  });
+
+  [
+    {
+      statusCode: 408,
```

Comment:
why does the test expect that we retry on 408 but not on 500 or 503?

Reply:
I inverted the implementation after changing... thanks :D

Reply:
removed 408